pkg/backup/backup_compaction.go

```diff
@@ -45,8 +48,81 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/uuid"
 	"github.com/cockroachdb/errors"
 	"github.com/gogo/protobuf/types"
+	"github.com/lib/pq"
 )
 
+var (
+	backupCompactionThreshold = settings.RegisterIntSetting(
+		settings.ApplicationLevel,
+		"backup.compaction.threshold",
+		"the maximum length of a backup chain before compaction occurs (0 to disable)",
+		0,
+		settings.WithVisibility(settings.Reserved),
+	)
+
+	backupCompactionWindow = settings.RegisterIntSetting(
+		settings.ApplicationLevel,
+		"backup.compaction.window",
+		"the number of backups to compact per compaction (must be greater than one and less than threshold)",
+		3,
+		settings.WithVisibility(settings.Reserved),
```

Comment:
nit: you can add the `IntWithMinimium` option and remove the min unit test and check. For example:

https://github.com/msbutler/cockroach/blob/butler-zero-stats/pkg/kv/kvserver/kvadmission/kvadmission.go#L136